xbmc/addons/kodi-addon-dev-kit/include/kodi/AddonBase.h

```diff
@@ -530,6 +530,9 @@ class ATTRIBUTE_HIDDEN CAddonBase
   static inline ADDON_STATUS ADDONBASE_CreateInstanceEx(int instanceType, const char* instanceID, KODI_HANDLE instance, KODI_HANDLE* addonInstance, KODI_HANDLE parent, const char* version)
   {
     ADDON_STATUS status = ADDON_STATUS_NOT_IMPLEMENTED;
+
+    *addonInstance = nullptr;
```

Comment:
Makes sense and thanks to find this fault.

Better to place it to this: https://github.com/xbmc/xbmc/blob/master/xbmc/addons/binary-addons/AddonDll.cpp#L332
with:
```cpp
  if (!CheckAPIVersion(instanceType))
    return ADDON_STATUS_PERMANENT_FAILURE;

  // Clear to nullptr for case addon not set it
  addonInstance = nullptr;

  KODI_HANDLE addonInstance;
  if (!m_interface.toAddon->create_instance_ex)
    status = m_interface.toAddon->create_instance(instanceType, instanceID.c_str(), instance, &addonInstance, parentInstance);
  else
    status = m_interface.toAddon->create_instance_ex(instanceType, instanceID.c_str(), instance, &addonInstance, parentInstance, kodi::addon::GetTypeVersion(instanceType));

```

Reply:
Agree completely, thanks!  Updated.